pympipool/__init__.py

```diff
@@ -1,15 +1,58 @@
+import os
 from ._version import get_versions
 from pympipool.mpi.executor import PyMPIExecutor
 
 try:  # The PyFluxExecutor requires flux-core to be installed.
     from pympipool.flux.executor import PyFluxExecutor
+
+    flux_installed = "FLUX_URI" in os.environ
 except ImportError:
+    flux_installed = False
     pass
 
 try:  # The PySlurmExecutor requires the srun command to be available.
     from pympipool.slurm.executor import PySlurmExecutor
+
+    slurm_installed = True
 except ImportError:
+    slurm_installed = False
     pass
 
+
 __version__ = get_versions()["version"]
 del get_versions
+
+
+class Executor:
+    def __new__(
+        cls,
+        max_workers,
```

Comment:
Is there a technical reason to avoid a default value for `max_workers`? The standard [`ThreadPoolExecutor`](https://docs.python.org/3/library/concurrent.futures.html#concurrent.futures.ThreadPoolExecutor) and [`ProcessPoolExecutor`](https://docs.python.org/3/library/concurrent.futures.html#concurrent.futures.ProcessPoolExecutor) have the same argument (which is a symmetry I find super helpful), but there this argument is optional.

This is a super minor thing, but it's just a point of polish to give some QoL. I ran into the issue back in contrib or workflow when I tried switching over to a pympipool executor from my silly custom one, but I had to go through and add `max_workers=1` to every executor instantiation because there was no more default. I like how much symmetry there already is with the standard tools and just want to maximize that where it won't cause technical headaches.